### PR TITLE
Pass environment to seed from manager

### PR DESCRIPTION
**Pull request by martenb**

### CODE REVIEW

File: src/Phinx/Migration/Manager.php [hunk 1]
```diff
@@ -865,7 +865,7 @@ protected function orderSeedsByDependencies(array $seeds)
      * @throws \InvalidArgumentException
      * @return \Phinx\Seed\AbstractSeed[]
      */
-    public function getSeeds()
+    public function getSeeds(string $environment)
```
Comment (dereuromark) [inline src/Phinx/Migration/Manager.php]: param docblock needs adjustment

**Review by dereuromark — APPROVED** [review]
Looks good to me

**Comment by MasterOdin** [comment]
@martenb could you rebase this against the `0.next` branch instead of `0.x` as while I'm not sure how many people have created custom managers, this is still editing the public interface, and so is a BC break.

As part of rebasing, please also add full type hints to all the new methods and such you've added.

**Comment by martenb** [comment]
@MasterOdin rebased

File: src/Phinx/Seed/AbstractSeed.php [hunk 2]
```diff
@@ -61,6 +66,24 @@ public function getDependencies(): array
...
+    {
+        $this->environment = $environment;
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getEnvironment()
```
Comment (MasterOdin) [inline src/Phinx/Seed/AbstractSeed.php]:
```suggestion
    public function getEnvironment(): string
```

File: src/Phinx/Seed/SeedInterface.php [hunk 3]
```diff
@@ -42,6 +42,20 @@ public function run(): void;
      */
     public function getDependencies(): array;
 
+    /**
+     * Sets the environment.
+     *
+     * @return \Phinx\Seed\SeedInterface
```
Comment (MasterOdin) [inline src/Phinx/Seed/SeedInterface.php]:
```suggestion
     * @return $this
```

File: src/Phinx/Seed/SeedInterface.php [hunk 4]
```diff
@@ -42,6 +42,20 @@ public function run(): void;
...
+     */
+    public function setEnvironment(string $environment);
+
+    /**
+     * Gets the environment.
+     *
+     * @return string
+     */
+    public function getEnvironment();
```
Comment (MasterOdin) [inline src/Phinx/Seed/SeedInterface.php]:
```suggestion
    public function getEnvironment(): string;
```

**Review by MasterOdin — CHANGES_REQUESTED** [review]
some minor nits, and then I think good to merge :+1:

**Review by MasterOdin — APPROVED** [review]

**Pull request merged**